### ISSUE TITLE
Expand Control Flow to component dom elements

### ISSUE THREAD

**Issue by n614cd** [angular/angular#53959]
### Which @angular/* package(s) are relevant/related to the feature request?

core

### Description

There are multiple CSS frameworks which depend upon specific DOM structures to be correctly applied. e.g. table -> th -> td
Introduction of any element in the model, breaks the CSS selector.
The new control flow syntax greatly reduces the amount of breakages I see. It would be nice to extend this further to eliminate the component in the DOM.

### Proposed solution

Create a new control flow, such as 
@component name (param1:value, param2:value)
This in theory would not include the component structure in the DOM.

### Alternatives considered

I hack the CSS frameworks.

**Comment by JoostK** [comment]
I don't quite understand what this is requesting, it seems like rendering components without host element?

**Comment by n614cd** [comment]
@JoostK 

Consider the following partial dom capture from Edge.
![image](https://github.com/angular/angular/assets/5581128/68542838-b9f6-4930-b43b-4aeea04adb5c)

There is an element in between form and the div. In this case, the element is a component I created called tms-form-field.
Inserting these additional elements breaks a css selector such as this one:
form div {
   background-color: grey;
}

Control flow, has eliminated the need for many intermediate dom nodes, such as ng-container used with an ngIf, ngFor... 

If possible, eliminating the component node in the dom tree along with the existing control flow I believe would eliminate the super majority of cases where a CSS framework, especially one designed by a UI/UX specialist is broken due to the selectors failing.

Tim

**Comment by JoostK** [comment]
Okay, that would be removing the host element in that case (#18877). Angular today is highly dependant on that element for features like host directives and host bindings/listeners, it's not just about component matching.

The new control flow syntax hasn't made a difference here; although `ng-container` is no longer present in the template, control flow still results in a view container for its insertion point (identical to what `ng-container` achieved).

**Comment by n614cd** [comment]
@JoostK 

Yes. When I searched before I did not find that issue. That thread had more use cases, but the same fundamental issue. The component element pollutes the HTML5 model, and causes problems with CSS, accessibility and other factors.
For me, accepting the fact you cannot have a host binding when the component element is not in the dom tree is a perfectly fine tradeoff.

Tim
